File: codes/java/chap01/answer_A03.java
import java.util.*;

class Main {
	public static void main(String[] args) {
		Scanner sc = new Scanner(System.in);

		// 入力
		int N = sc.nextInt();
		int K = sc.nextInt();
		int[] P = new int[N + 1];
		int[] Q = new int[N + 1];
		for (int i = 1; i <= N; i++) P[i] = sc.nextInt();
		for (int i = 1; i <= N; i++) Q[i] = sc.nextInt();

		// 全探索（Answer は「合計が K になる選び方が見つかったか」を示す）
		boolean Answer = false;
		for (int x = 1; x <= N; x++) {
			for (int y = 1; y <= N; y++) {
				if (P[x] + Q[y] == K) Answer = true;
			}
		}

		// 出力
		if (Answer == true) System.out.println("Yes");
		else System.out.println("No");
	}
};

File: codes/cpp/chap01/answer_A03.cpp
#include <iostream>
using namespace std;

int N, K;
int P[109], Q[109];
bool Answer = false;

int main() {
	// 入力
	cin >> N >> K;
	for (int i = 1; i <= N; i++) cin >> P[i];
	for (int i = 1; i <= N; i++) cin >> Q[i];

	// 全探索（Answer は「合計が K になる選び方が見つかったか」を示す）
	for (int x = 1; x <= N; x++) {
		for (int y = 1; y <= N; y++) {
			if (P[x] + Q[y] == K) Answer = true;
		}
	}

	// 出力
	if (Answer == true) cout << "Yes" << endl;
	else cout << "No" << endl;
	return 0;
}
